lib/elixir/lib/code.ex

```diff
@@ -1197,6 +1197,188 @@ defmodule Code do
     :elixir.string_to_quoted!(to_charlist(string), line, column, file, opts)
   end
 
+  @doc """
+  Converts the given string to its quoted form and a list of comments.
+
+  This function is useful when performing textual changes to the source code,
+  while preserving information like comments and literals position.
+
+  Returns `{:ok, quoted_form, comments}` if it succeeds,
+  `{:error, {line, error, token}}` otherwise.
+
+  The AST returned differs from regular AST in that literals are wrapped in
+  a block to preserve their metadata. For example, the literal `"foo"` will be
+  converted to `{:__block__, [line: 1, delimiter: "\"", token: "\"foo\""], ["foo"]}`.
```

Comment:
I don't think we should do this by default. It should be an option, like you pass to `string_to_quoted`, and we make the comments work even without this?

Reply:
So, the function would return regular ast + comments, and for the formatter use cases we would still pass the `literal_encoder` option?

Reply:
Yes. literal_encoder, token_metadata, etc should all be explicitly given (as in string_to_quoted). We will only set preserve_comments (which is private).

Reply:
Done!